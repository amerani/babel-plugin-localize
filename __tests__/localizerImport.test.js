const babel = require('@babel/core');
const plugin = require('../src/index');

const example = `
export default () => (<p>hello world</p>);
`;

test('should import localizer', () => {
  const { code } = babel.transform(example, { plugins: [plugin] });
  expect(code).toMatchSnapshot();
});

test('should import default localizer', () => {
  const { code } = babel.transform(example, {
    plugins: [[plugin, {
      localizerBinding: 'default',
    }]],
  });
  expect(code).toMatchSnapshot();
});
