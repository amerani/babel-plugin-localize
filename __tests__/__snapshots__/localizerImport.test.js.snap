// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should import default localizer 1`] = `
"import localize from \\"./localizer\\";
export default (() => <p>{localize(\\"1\\")}</p>);
export const localizeKeyMap = {
  1: \\"hello world\\"
};"
`;

exports[`should import localizer 1`] = `
"import { localize } from \\"./localizer\\";
export default (() => <p>{localize(\\"0\\")}</p>);
export const localizeKeyMap = {
  0: \\"hello world\\"
};"
`;
